packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -167,6 +167,31 @@ class FileAccessLog extends AccessLog {
   }
 }
 
+/**
+ * Represents the properties needed to define client policy
+ */
+export interface TlsClientPolicy {
```

Comment:
Let's move this interface to the same file that defines `TlsValidationContext`, and the remaining classes/interfaces there.

Reply:
Could you advise the intention for moving `TlsClientPolicy` interface? If this is happening, I am wondering if `BackendDefaults` should also move too,

```
export interface BackendDefaults {
  /**
   * Client policy for backend defaults
   *
   * @default - none
   */
  readonly tlsClientPolicy?: TlsClientPolicy;
}
```

Reply:
> Could you advise the intention for moving TlsClientPolicy interface?

I don't like files with generic names like `shared-interfaces`. Over time, the become dumping grounds for things, and they get very long. Things should be grouped logically - things are logically close together, like `TlsClientPolicy` and `TlsValidation`, should be in one place.

> If this is happening, I am wondering if BackendDefaults should also move too/

Probably, but this is unrelated to your change, so feel free to keep them here for this PR.

Reply:
Thanks Adam! It makes sense that overall motivation is to avoid using generic file. Will move `TlsClientPolicy` interface as advised.